loadClass throws CNFE if findClass via bootstrapClassLoader not succeed

JasonFengJ9 opened the pull request:
`loadClass(className)` should throw `ClassNotFoundException` if `findClass` via `bootstrapClassLoader` returns `null`;
`Class.forName(className)` in `java.base` can't load the classes outside of `java.base` module;
Add a test to verify the `classloader` of the class loaded via `automatic` module path, rename `classloadertest.xml` to
`enableAssertionStatusTest.xml`, add `autoModuleClassloaderTest.xml`, and update `playlist.xml`.

closes https://github.com/eclipse-openj9/openj9/issues/14975
This reverts https://github.com/eclipse-openj9/openj9/pull/14689

fyi @tajila 

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

pshipton commented:
jenkins compile alinux64 jdk8,jdk11

pshipton commented:
Tested via grinder https://openj9-jenkins.osuosl.org/view/Test/job/Grinder/800

Pls change the name of the test in the playlist.xml. It's called cmdLineTester_EnableAssertionStatusTest

JasonFengJ9 commented:
> Pls change the name of the test in the playlist.xml. It's called cmdLineTester_EnableAssertionStatusTest

Renamed `classloadertest.xml` to `enableAssertionStatusTest.xml`, added `autoModuleClassloaderTest.xml`, and updated `playlist.xml`.

pshipton commented:
New grinder looks good https://openj9-jenkins.osuosl.org/view/Test/job/Grinder/801

@llxia any concerns from the test side. The grinder with dynamic compile worked, but not sure if any updates are needed for a new test.

llxia reviewed (approved):
LGTM
